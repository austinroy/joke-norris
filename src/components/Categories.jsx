import React, { Component } from 'react';
import SearchBar from './SearchBar';
import axios from 'axios';

class Categories extends Component {
  state = {
      categories : []
  }
  componentDidMount = () => {
    axios.get('https://api.chucknorris.io/jokes/categories')
    .then(res =>{
        if(res.status === 200) {
            const categories = res.data;
            this.setState({categories : categories})
        }
      }
    )
  }

  renderCategory = (category, index) => {
    const categoryUrl = `/categories/${category}`
      return(
        <div className="category" key={index}>
          <a href={categoryUrl} className="category">
              {category}
          </a>
        </div>
    )
  }

  mapCategories = categories => {
    if (categories.length > 0) {
      return categories.map((category, index) => {
        return this.renderCategory(category, index)
      })
    } else {
      return (
        <div className="not-loaded">
          Categories Not Loaded
        </div>
      )
    }
  }
  
  render () {
    const {categories} = this.state;
    console.log(categories);
    return(
      <div>
        <SearchBar />
        <h3>Categories</h3>
        <div className="categories">
          { this.mapCategories(categories)}
        </div>
      </div>
    )
  }
}

export default Categories;