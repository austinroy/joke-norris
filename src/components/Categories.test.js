import { configure } from 'enzyme';
import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
// import { shallow, mount, render } from 'enzyme';
import { Categories } from './Categories';
import { render, cleanup } from '@testing-library/react';

configure({ adapter: new Adapter() });

const props = {
  fetchCategories: jest.fn(),
  loading: false,
  categories: [
    'explicit',
    'dev',
    'movie',
    'food',
    'celebrity',
    'science',
    'sport',
    'political',
    'religion',
    'animal',
    'history',
    'music',
    'travel',
    'career',
    'money',
    'fashion'
  ]
};

afterEach(cleanup);

const propsLoading = {
  fetchCategories: jest.fn(),
  loading: true
};

test('renders category cards', () => {
  const container = render(<Categories {...props} />);
  expect(container).toMatchSnapshot();
});

test('renders a loader', () => {
  const container = render(<Categories {...propsLoading} />);
  expect(container).toMatchSnapshot();
});
