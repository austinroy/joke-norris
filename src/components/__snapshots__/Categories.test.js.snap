// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a loader 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="u-center-text"
      >
        <div
          class="ui grey medium header"
        >
          Categories
        </div>
        <p
          class="u-center-text"
        >
          Chuck Norris demands you choose a category
        </p>
        <div
          class="categories"
        >
          <div
            class="ui active inline loader"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="u-center-text"
    >
      <div
        class="ui grey medium header"
      >
        Categories
      </div>
      <p
        class="u-center-text"
      >
        Chuck Norris demands you choose a category
      </p>
      <div
        class="categories"
      >
        <div
          class="ui active inline loader"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`renders category cards 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="u-center-text"
      >
        <div
          class="ui grey medium header"
        >
          Categories
        </div>
        <p
          class="u-center-text"
        >
          Chuck Norris demands you choose a category
        </p>
        <div
          class="categories"
        >
          <div
            class="categories__Card"
          >
            <a
              class="ui green raised card"
              href="/categories/explicit"
            >
              <div
                class="content"
              >
                <div
                  class="header"
                  style="text-transform: capitalize;"
                >
                  explicit
                </div>
              </div>
            </a>
          </div>
          <div
            class="categories__Card"
          >
            <a
              class="ui green raised card"
              href="/categories/dev"
            >
              <div
                class="content"
              >
                <div
                  class="header"
                  style="text-transform: capitalize;"
                >
                  dev
                </div>
              </div>
            </a>
          </div>
          <div
            class="categories__Card"
          >
            <a
              class="ui green raised card"
              href="/categories/movie"
            >
              <div
                class="content"
              >
                <div
                  class="header"
                  style="text-transform: capitalize;"
                >
                  movie
                </div>
              </div>
            </a>
          </div>
          <div
            class="categories__Card"
          >
            <a
              class="ui green raised card"
              href="/categories/food"
            >
              <div
                class="content"
              >
                <div
                  class="header"
                  style="text-transform: capitalize;"
                >
                  food
                </div>
              </div>
            </a>
          </div>
          <div
            class="categories__Card"
          >
            <a
              class="ui green raised card"
              href="/categories/celebrity"
            >
              <div
                class="content"
              >
                <div
                  class="header"
                  style="text-transform: capitalize;"
                >
                  celebrity
                </div>
              </div>
            </a>
          </div>
          <div
            class="categories__Card"
          >
            <a
              class="ui green raised card"
              href="/categories/science"
            >
              <div
                class="content"
              >
                <div
                  class="header"
                  style="text-transform: capitalize;"
                >
                  science
                </div>
              </div>
            </a>
          </div>
          <div
            class="categories__Card"
          >
            <a
              class="ui green raised card"
              href="/categories/sport"
            >
              <div
                class="content"
              >
                <div
                  class="header"
                  style="text-transform: capitalize;"
                >
                  sport
                </div>
              </div>
            </a>
          </div>
          <div
            class="categories__Card"
          >
            <a
              class="ui green raised card"
              href="/categories/political"
            >
              <div
                class="content"
              >
                <div
                  class="header"
                  style="text-transform: capitalize;"
                >
                  political
                </div>
              </div>
            </a>
          </div>
          <div
            class="categories__Card"
          >
            <a
              class="ui green raised card"
              href="/categories/religion"
            >
              <div
                class="content"
              >
                <div
                  class="header"
                  style="text-transform: capitalize;"
                >
                  religion
                </div>
              </div>
            </a>
          </div>
          <div
            class="categories__Card"
          >
            <a
              class="ui green raised card"
              href="/categories/animal"
            >
              <div
                class="content"
              >
                <div
                  class="header"
                  style="text-transform: capitalize;"
                >
                  animal
                </div>
              </div>
            </a>
          </div>
          <div
            class="categories__Card"
          >
            <a
              class="ui green raised card"
              href="/categories/history"
            >
              <div
                class="content"
              >
                <div
                  class="header"
                  style="text-transform: capitalize;"
                >
                  history
                </div>
              </div>
            </a>
          </div>
          <div
            class="categories__Card"
          >
            <a
              class="ui green raised card"
              href="/categories/music"
            >
              <div
                class="content"
              >
                <div
                  class="header"
                  style="text-transform: capitalize;"
                >
                  music
                </div>
              </div>
            </a>
          </div>
          <div
            class="categories__Card"
          >
            <a
              class="ui green raised card"
              href="/categories/travel"
            >
              <div
                class="content"
              >
                <div
                  class="header"
                  style="text-transform: capitalize;"
                >
                  travel
                </div>
              </div>
            </a>
          </div>
          <div
            class="categories__Card"
          >
            <a
              class="ui green raised card"
              href="/categories/career"
            >
              <div
                class="content"
              >
                <div
                  class="header"
                  style="text-transform: capitalize;"
                >
                  career
                </div>
              </div>
            </a>
          </div>
          <div
            class="categories__Card"
          >
            <a
              class="ui green raised card"
              href="/categories/money"
            >
              <div
                class="content"
              >
                <div
                  class="header"
                  style="text-transform: capitalize;"
                >
                  money
                </div>
              </div>
            </a>
          </div>
          <div
            class="categories__Card"
          >
            <a
              class="ui green raised card"
              href="/categories/fashion"
            >
              <div
                class="content"
              >
                <div
                  class="header"
                  style="text-transform: capitalize;"
                >
                  fashion
                </div>
              </div>
            </a>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="u-center-text"
    >
      <div
        class="ui grey medium header"
      >
        Categories
      </div>
      <p
        class="u-center-text"
      >
        Chuck Norris demands you choose a category
      </p>
      <div
        class="categories"
      >
        <div
          class="categories__Card"
        >
          <a
            class="ui green raised card"
            href="/categories/explicit"
          >
            <div
              class="content"
            >
              <div
                class="header"
                style="text-transform: capitalize;"
              >
                explicit
              </div>
            </div>
          </a>
        </div>
        <div
          class="categories__Card"
        >
          <a
            class="ui green raised card"
            href="/categories/dev"
          >
            <div
              class="content"
            >
              <div
                class="header"
                style="text-transform: capitalize;"
              >
                dev
              </div>
            </div>
          </a>
        </div>
        <div
          class="categories__Card"
        >
          <a
            class="ui green raised card"
            href="/categories/movie"
          >
            <div
              class="content"
            >
              <div
                class="header"
                style="text-transform: capitalize;"
              >
                movie
              </div>
            </div>
          </a>
        </div>
        <div
          class="categories__Card"
        >
          <a
            class="ui green raised card"
            href="/categories/food"
          >
            <div
              class="content"
            >
              <div
                class="header"
                style="text-transform: capitalize;"
              >
                food
              </div>
            </div>
          </a>
        </div>
        <div
          class="categories__Card"
        >
          <a
            class="ui green raised card"
            href="/categories/celebrity"
          >
            <div
              class="content"
            >
              <div
                class="header"
                style="text-transform: capitalize;"
              >
                celebrity
              </div>
            </div>
          </a>
        </div>
        <div
          class="categories__Card"
        >
          <a
            class="ui green raised card"
            href="/categories/science"
          >
            <div
              class="content"
            >
              <div
                class="header"
                style="text-transform: capitalize;"
              >
                science
              </div>
            </div>
          </a>
        </div>
        <div
          class="categories__Card"
        >
          <a
            class="ui green raised card"
            href="/categories/sport"
          >
            <div
              class="content"
            >
              <div
                class="header"
                style="text-transform: capitalize;"
              >
                sport
              </div>
            </div>
          </a>
        </div>
        <div
          class="categories__Card"
        >
          <a
            class="ui green raised card"
            href="/categories/political"
          >
            <div
              class="content"
            >
              <div
                class="header"
                style="text-transform: capitalize;"
              >
                political
              </div>
            </div>
          </a>
        </div>
        <div
          class="categories__Card"
        >
          <a
            class="ui green raised card"
            href="/categories/religion"
          >
            <div
              class="content"
            >
              <div
                class="header"
                style="text-transform: capitalize;"
              >
                religion
              </div>
            </div>
          </a>
        </div>
        <div
          class="categories__Card"
        >
          <a
            class="ui green raised card"
            href="/categories/animal"
          >
            <div
              class="content"
            >
              <div
                class="header"
                style="text-transform: capitalize;"
              >
                animal
              </div>
            </div>
          </a>
        </div>
        <div
          class="categories__Card"
        >
          <a
            class="ui green raised card"
            href="/categories/history"
          >
            <div
              class="content"
            >
              <div
                class="header"
                style="text-transform: capitalize;"
              >
                history
              </div>
            </div>
          </a>
        </div>
        <div
          class="categories__Card"
        >
          <a
            class="ui green raised card"
            href="/categories/music"
          >
            <div
              class="content"
            >
              <div
                class="header"
                style="text-transform: capitalize;"
              >
                music
              </div>
            </div>
          </a>
        </div>
        <div
          class="categories__Card"
        >
          <a
            class="ui green raised card"
            href="/categories/travel"
          >
            <div
              class="content"
            >
              <div
                class="header"
                style="text-transform: capitalize;"
              >
                travel
              </div>
            </div>
          </a>
        </div>
        <div
          class="categories__Card"
        >
          <a
            class="ui green raised card"
            href="/categories/career"
          >
            <div
              class="content"
            >
              <div
                class="header"
                style="text-transform: capitalize;"
              >
                career
              </div>
            </div>
          </a>
        </div>
        <div
          class="categories__Card"
        >
          <a
            class="ui green raised card"
            href="/categories/money"
          >
            <div
              class="content"
            >
              <div
                class="header"
                style="text-transform: capitalize;"
              >
                money
              </div>
            </div>
          </a>
        </div>
        <div
          class="categories__Card"
        >
          <a
            class="ui green raised card"
            href="/categories/fashion"
          >
            <div
              class="content"
            >
              <div
                class="header"
                style="text-transform: capitalize;"
              >
                fashion
              </div>
            </div>
          </a>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
